graviti/openapi/policy.py

```diff
@@ -131,6 +216,61 @@ def get_object(self, key: str, _allow_retry: bool = True) -> UserResponse:
                 return self.get_object(key, False)
             raise error from None
 
+    def put_object(self, file: File, pbar: tqdm, _allow_retry: bool = True) -> None:
+        """Put the object to OSS.
+
+        Arguments:
+            file: The local file to upload.
+            pbar: The process bar for uploading binary files.
+            _allow_retry: Whether requesting the put policy again is allowed.
+
+        Raises:
+            ResponseError: If post response error.
+
+        """
+        key = file.key
+        path = file.path
+        policy = self._init_put_policy()
+        date = _from_datetime_to_gmt(datetime.now(timezone.utc))
+        verb = "PUT"
+
+        authorization = self._get_authorization(policy, verb, date, key)
+        mime_type = mimetypes.guess_type(path)[0]
+        headers = {
+            "Authorization": authorization,
+            "x-oss-security-token": policy["SecurityToken"],
+            "Date": date,
+            "Content-Type": mime_type,
+        }
+        url = f"https://{policy['bucket']}.{policy['endpoint']}/{key}"
+
+        try:
+            with path.open("rb") as fp:
+                do(verb, url, headers=headers, data=fp)
```

Comment:
do we need `MultipartEncoder` to upload file?

Reply:
no, here we use putObject method instead of postObject method